tls/extensions/s2n_extension_list.h

```diff
@@ -23,11 +23,13 @@
 typedef struct {
     uint16_t extension_type;
     struct s2n_blob extension;
+    uint16_t wire_index;
 } s2n_parsed_extension;
 
 typedef struct {
     s2n_parsed_extension parsed_extensions[S2N_PARSED_EXTENSIONS_COUNT];
     struct s2n_blob raw; /* Needed by some ClientHello APIs */
+    uint16_t count;
```

Comment:
Do we need `uint16_t` for these index/count values, isn't `uint8_t` sufficient?

Reply:
uint8_t is not sufficient. The max theoretically possible (if all extensions were 0-length) is UINT16_MAX (max size of the extensions list) / 4 (2 bytes id, 2 bytes length), which is still > UINT8_MAX.